src/backend/distributed/shared_library_init.c

```diff
@@ -1238,6 +1238,16 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.max_cached_connection_lifetime",
+		gettext_noop("Sets the maximum lifetime of cached connections to worker nodes."),
```

Comment:
maybe avoid "worker nodes" as we could have the cached connection to the coordinator as well.

Also, pgbouncer seems to use 3600 seconds, though with Citus it is better to have a smaller default given the number of shards and potential of sequential execution, I  guess?

Reply:
10 minutes is based on what we use(d) in the managed service.